### ISSUE TITLE
auto sell junk issue

### ISSUE THREAD

**Issue by rml9ty** [arkayenro/arkinventory#1383]
issue: junk is not sold automatically (using version v3.09.41 on retail)

those are my settings: https://imgur.com/ZMt13jg

why isn't anything sold automatically? I would appreciate if you took a quick look. :)

**Comment by arkayenro** [comment]
the settings look ok.  is this happening for all vendors, or just one?  there are some vendors that wont buy stuff.

you have list enabled so is it listing each item and just not selling them? or not listing them at all?

**Comment by rml9ty** [comment]
tried different vendors. they definitely do usually buy stuff in general, the vendors I'm at that is.

it neither lists stuff, nor does it sell it. basically nothing happens at all.

**Comment by arkayenro** [comment]
your screenshot is ok, those settings are global

there is code to try and work out if a vendor is one the few that wont take stuff but typically that fails more than it works.

the only reason i can think of for it not auto selling is because while its categorised as junk the junk category doesnt have auto-sell enabled

can you go into edit mode and click on the one of the junk items, then mouseover its category.  make sure the category its assigned to has auto sell enabled, just because its junk doesnt mean it will auto sell as well.

**Comment by rml9ty** [comment]
hmm.
just made a quick video / gif, which may be more telling than a screenshot. https://gfycat.com/thornyanydodobird

where did I go wrong?

**Comment by arkayenro** [comment]
it looks ok so the only thing left i can think of is that you have scrap, selljunk, or reagentrestoker, loaded.  if any of those are loaded then it will leave the junk selling part up to them.

if that was the case though you should see a warning message  about junk selling being disabled when you first enter the world

**Comment by rml9ty** [comment]
nah, I tried it only having arkinventory enabled, just to make sure.

weirdly, this even is a completely new setup. decided to setup all from scratch for SL. hm.

**Comment by rml9ty** [comment]
okay, I've only had ArkInventory enabled. no other addons.
I made a completely new setup, from scratch. deleted all WTF files related to ArkInventory.
even grey junk does not sell, nor is it listed in chat with sell junk test mode. 

this just aint right ^_^

**Comment by rml9ty** [comment]
man, I got it. and it's so... so... unexpected. if you have "auto open at merchant/vendor" disabled it does not work, even though you talk to one. -.-
finally....

**Comment by rml9ty** [comment]
is that something you could change? the event triggering by talking to a vendor manually? (:

**Comment by arkayenro** [comment]
at the moment it wants the bag open before the junksell code triggers, but i suppose it can trigger just off the merchant open
